src/sentry/workflow_engine/processors/action.py

```diff
@@ -127,11 +130,122 @@ def filter_recently_fired_workflow_actions(
     actions_without_statuses_ids = {action.id for action in actions_without_statuses}
     filtered_actions = actions.filter(id__in=actions_to_include | actions_without_statuses_ids)
 
+    # dual write to WorkflowActionGroupStatus
+    filter_recently_fired_workflow_actions(filtered_action_groups, event_data)
+
     create_workflow_fire_histories(filtered_actions, event_data)
 
     return filtered_actions
 
 
+def get_workflow_group_action_statuses(
+    action_to_workflows_ids: dict[int, set[int]], group: Group
+) -> dict[int, list[WorkflowActionGroupStatus]]:
+    all_statuses = WorkflowActionGroupStatus.objects.filter(
+        group=group, action_id__in=action_to_workflows_ids.keys()
+    )
```

Comment:
wasn't sure how to query for specific action+workflow combos, so opted for iterating through a big query